octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -271,124 +308,699 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
+                                          check_completeness: bool = True, **kwargs: dict) -> typing.Optional[dict]:
+        if self.client.has.get('fetchOrder'):
+            try:
+                with self.error_describer():
+                    params = kwargs.pop("params", {})
+                    if params := self.exchange_manager.exchange.custom_get_order_stop_params(order_id, params):
+                        if order := await self.client.fetch_order(order_id, symbol, params=params):
+                            await self.exchange_manager.exchange.parse_order(order,
+                                                                             check_completeness=check_completeness)
+            except ccxt.OrderNotFound:
+                # some exchanges are throwing this error when an order is cancelled
+                pass
+            except Exception as e:
+                self.logger.exception(e, True, "Failed to get order using get_order_using_stop_params")
+        return None
+
+    async def get_order_from_open_and_closed_orders(self, order_id: str, symbol: str = None,
+                                                    check_completeness: bool = True, **kwargs: dict
+                                                    ) -> typing.Optional[dict]:
+        for order in await self.get_open_orders(symbol, check_completeness=check_completeness, **kwargs):
+            if order[ecoc.ID.value] == order_id:
+                return order
+        for order in await self.get_closed_orders(symbol, check_completeness=check_completeness, **kwargs):
+            if order[ecoc.ID.value] == order_id:
+                return order
         return None  # OrderNotFound
 
-    async def get_all_orders(self, symbol: str = None, since: int = None,
-                             limit: int = None, **kwargs: dict) -> list:
-        if self.client.has['fetchOrders']:
+    async def get_all_orders(self, symbol: str = None, since: int = None, limit: int = None,
+                             check_completeness: bool = True, **kwargs: dict) -> list:
+        limit = self.cut_order_pagination_limit(limit)
+        defined_methods = self.connector_config.GET_ALL_ORDERS_METHODS
+        orders = []
+        if self.get_all_orders_default.__name__ in defined_methods:
+            orders += await self.get_all_orders_default(symbol=symbol, since=since, limit=limit,
+                                                        check_completeness=check_completeness, kwargs=kwargs)
+        if self.get_all_stop_orders_using_stop_loss_params.__name__ in defined_methods:
+            orders += await self.get_all_stop_orders_using_stop_loss_params(symbol=symbol, since=since, limit=limit,
+                                                                            check_completeness=check_completeness,
+                                                                            kwargs=kwargs)
```

Comment:
Can there be duplicates?

Reply:
potentially, but not on bybit. 
And if the order id is the same it shouldnt cause damage as the last duplicate would overwrite the previous data